drivers/gles3/rasterizer_scene_gles3.h

```diff
@@ -508,7 +509,16 @@ class RasterizerSceneGLES3 : public RendererSceneRender {
 
 		struct SortByReverseDepthAndPriority {
 			_FORCE_INLINE_ bool operator()(const GeometryInstanceSurface *A, const GeometryInstanceSurface *B) const {
-				return (A->sort.priority == B->sort.priority) ? (A->owner->depth > B->owner->depth) : (A->sort.priority < B->sort.priority);
+				if (A->sort.priority == B->sort.priority) {
+					if (Math::abs(A->owner->depth - B->owner->depth) < 0.001) {
```

Comment:
```suggestion
					if (A->owner->depth == B->owner->depth) {
```
When these passes are on the same MeshInstance3D the `owner` will be the same meaning that the `depth` values will be exactly the same. Accordingly we can use ``==`` instead of the approximate equals which is much faster.